src/workerd/api/pyodide/pyodide.c++

```diff
@@ -105,22 +93,39 @@ kj::Array<kj::String> _getWorkerFiles(
   return builder.releaseAsArray();
 }
 
-kj::HashSet<kj::String> _getNames(kj::ArrayPtr<kj::String> names) {
+kj::HashSet<kj::String> PythonModuleInfo::getWorkerModuleSet() {
   auto result = kj::HashSet<kj::String>();
-  for (auto& name: names) {
-    if (!name.endsWith(".py")) {
+  const auto vendor = "vendor/"_kj;
+  const auto dotpy = ".py"_kj;
+  const auto dotso = ".so"_kj;
+  for (auto& name_: names) {
+    kj::StringPtr name = name_;
+    if (name.startsWith(vendor)) {
+      name = name.slice(vendor.size());
+    }
+    auto firstSlash = name.findFirst('/');
```

Comment:
Do we have anything in kj or C++ stdlib that might make it easier to do this file path parsing?

I guess we don't have to worry about Linux vs. Windows dir separator differences here, so it's not a big deal.

Reply:
Well the logic here isn't so complicated I don't think.